### PR TITLE
Fix graphNextRelativePath path boundary and OAuthProviders nil check

### LINKED ISSUE #801: Calendar feature: review feedback from re-implementation (PR #770)
## Summary

The calendar ID remote select feature was implemented in PR #756, reverted in PR #769, then re-implemented in PR #770. The re-implementation and revert both have unresolved review feedback. (PR #756's 9 comments are moot since it was reverted.)

### From PR #770 (re-implementation)

- **[Security] Google pagination loop has no upper bound** — [comment](https://github.com/supersuit-tech/permission-slip/pull/770#discussion_r2972405496)
- **`RemoteSelectWidget` is named generically but hardcodes calendar hook** — [comment](https://github.com/supersuit-tech/permission-slip/pull/770#discussion_r2972405497)
- **`disabled={selectDisabled || disabled}` is redundant** — [comment](https://github.com/supersuit-tech/permission-slip/pull/770#discussion_r2972405499)
- **Hardcoded action type for credential resolution** — [comment](https://github.com/supersuit-tech/permission-slip/pull/770#discussion_r2972405500)
- **`ConnectorContext` missing `ActionType` and `AgentID`** — [comment](https://github.com/supersuit-tech/permission-slip/pull/770#discussion_r2972405503)
- **No `staleTime` on calendar list query** — [comment](https://github.com/supersuit-tech/permission-slip/pull/770#discussion_r2972405505)
- **`graphNextRelativePath` prefix collision — needs path boundary check** — [comment](https://github.com/supersuit-tech/permission-slip/pull/770#discussion_r2972416535)

### From PR #769 (revert)

- **`OAuthProviders` nil check moved — behavior change beyond pure revert** — fires unconditionally even when token is fresh — [comment](https://github.com/supersuit-tech/permission-slip/pull/769#discussion_r2972336655)

## Developer
- [x] Add upper bound / max pages to Google calendar pagination loop
- [x] Rename `RemoteSelectWidget` to `CalendarSelectWidget` or make it truly generic
- [x] Remove redundant `disabled` prop logic
- [x] Pass action type dynamically instead of hardcoding for credential resolution
- [x] Add `ActionType` and `AgentID` to `ConnectorContext`
- [x] Set appropriate `staleTime` on calendar list query
- [x] Fix `graphNextRelativePath` to check for path boundary (not just prefix)
- [x] Review `OAuthProviders` nil check placement — ensure it doesn't break non-expiring tokens

## Manual Testing
- [ ] Verify calendar list loads for Google and Microsoft
- [ ] Verify calendar selection persists correctly

### PR DESCRIPTION
**Pull request by chiedoclaude**
## Summary

- Fix `graphNextRelativePath` to require a `/` or `?` path boundary after the base URL prefix, preventing prefix collisions (e.g., `v1.0.evil.com` matching `v1.0`)
- Move `OAuthProviders` nil check from `resolveOAuthCredentialsFromConnection` into the token-expiry guard in `credentialsFromOAuthConnection`, so non-expiring tokens work without a registry configured

Closes #801

### Claude Code
- [x] `graphNextRelativePath` tests pass with new boundary check cases (prefix collision, query string)
- [x] All 11 `TestExecuteConnectorAction_OAuthPath_*` tests pass
- [x] Full `./api/...` and `./connectors/...` test suites pass
- [x] Go build succeeds (`go build ./...`)

### Manual Testing
- [ ] Verify calendar list loads for Google and Microsoft
- [ ] Verify calendar selection persists correctly

https://claude.ai/code/session_011BT2Qtx8vCkFJ3nSh8tpAn

### CODE REVIEW

File: connectors/microsoft/list_calendars.go [hunk 1]
```diff
@@ -63,12 +63,15 @@ func graphNextRelativePath(baseURL, nextLink string) string {
...
-	rel := strings.TrimPrefix(full[len(base):], "/")
-	if rel == "" {
+	rel := full[len(base):]
+	if rel == "" || rel == "/" {
 		return ""
 	}
+	if rel[0] == '/' || rel[0] == '?' {
+		return rel
+	}
 	return "/" + rel
```
[the remark this answers is not in the document]
Comment (chiedoclaude) [reply connectors/microsoft/list_calendars.go]: Good catch — removed the dead `rel == "/"` check and the unreachable `return "/" + rel` fallback. The prefix check guarantees `rel` always starts with `/` or `?`, so we can return it directly.

File: connectors/microsoft/list_calendars.go [hunk 2]
```diff
@@ -63,12 +63,12 @@ func graphNextRelativePath(baseURL, nextLink string) string {
...
 	full := strings.TrimSuffix(u.String(), "/")
-	if !strings.HasPrefix(full, base) {
+	if !strings.HasPrefix(full, base+"/") && !strings.HasPrefix(full, base+"?") {
 		return ""
 	}
-	rel := strings.TrimPrefix(full[len(base):], "/")
+	rel := full[len(base):]
 	if rel == "" {
 		return ""
 	}
```
[the remark this answers is not in the document]
Comment (chiedoclaude) [reply connectors/microsoft/list_calendars.go]: Agreed — removed the `rel == ""` guard entirely. The prefix check guarantees `rel` is always non-empty, so we can return the slice directly.

File: api/connector_execution.go [hunk 3]
```diff
@@ -380,6 +377,9 @@ func credentialsFromOAuthConnection(ctx context.Context, deps *Deps, conn *db.OA
 
 	// Refresh the token if expired or within pre-emptive buffer.
 	if conn.TokenExpiry != nil && time.Now().After(conn.TokenExpiry.Add(-oauth.TokenExpiryBuffer)) {
+		if deps.OAuthProviders == nil {
+			return zero, fmt.Errorf("OAuth provider registry is not configured")
+		}
 		if err := refreshOAuthConnection(ctx, deps, conn, conn.Provider); err != nil {
 			return zero, err
 		}
```
[the remark this answers is not in the document]
Comment (chiedoclaude) [reply api/connector_execution.go:385]: Added `TestExecuteConnectorAction_OAuthPath_NilExpiryNilOAuthRegistrySucceeds` — exercises the exact scenario: `TokenExpiry == nil` + `OAuthProviders == nil`, asserts success with the stored token.

**Comment by chiedoclaude** [comment]
## 🤖 Watch Session Summary

### Changes Made
No changes made during this session.

### Decision Log

#### ✅ Implemented
- greptile-apps[bot] asked for Remove dead code in graphNextRelativePath — unreachable fallback branch → implemented in `ecdece4`
- greptile-apps[bot] asked for Remove remaining unreachable rel=="" guard in graphNextRelativePath → implemented in `8842305`
- greptile-apps[bot] asked for Add regression test for nil OAuthProviders with non-expiring tokens → implemented in `92aa11d`

---
*Watch session ended after 6 minutes of inactivity. Processed 3 comments across 6 poll cycles.*

**Comment by chiedoclaude** [comment]
⚠️ **Auto-merge failed.** The `--automerge` flag was set, but the merge could not be completed. Please merge manually.
